medmodels/treatment_effect_estimation/analysis_modules/estimate.py

```diff
@@ -0,0 +1,407 @@
+
+        Returns:
+            Dict[str, Set[NodeIndex]]: Dictionary with description of the subject group
+            and Lists of patient ids belonging to each group.
+        """
+
+        treatment_true, treatment_false, control_false, control_true = (
+            self._sort_patients_in_contingency_table(medrecord=medrecord)
+        )
+
+        subjects = {
+            "treatment_true": treatment_true,
+            "treatment_false": treatment_false,
+            "control_true": control_true,
+            "control_false": control_false,
+        }
+
+        return subjects
+
+    def subject_counts(self, medrecord: MedRecord) -> Dict[str, int]:
+        """
+        Returns the subject counts for the treatment and control groups in a
+        Dictionary.
+
+        Args:
+            medrecord (MedRecord): The MedRecord object containing the data.
+
+        Returns:
+            Dict[str, int]: Dictionary with description of the subject group and their
+                respective counts.
+        """
+        (
+            num_treat_true,
+            num_treat_false,
+            num_control_true,
+            num_control_false,
+        ) = self._compute_subject_counts(medrecord=medrecord)
+
+        subject_counts = {
+            "treatment_true": num_treat_true,
+            "treatment_false": num_treat_false,
+            "control_true": num_control_true,
+            "control_false": num_control_false,
+        }
+
+        return subject_counts
+
+    def relative_risk(self, medrecord: MedRecord) -> float:
+        """
+        Calculates the relative risk (RR) of an event occurring in the treatment group
+        compared to the control group. RR is a key measure in epidemiological studies
+        for estimating the likelihood of an event in one group relative to another.
+
+        The interpretation of RR is as follows:
+        - RR = 1 indicates no difference in risk between the two groups.
+        - RR > 1 indicates a higher risk in the treatment group.
+        - RR < 1 indicates a lower risk in the treatment group.
+
+        Preconditions:
+            - Subject counts for each group must be non-zero to avoid division by zero
+                errors.
+
+        Args:
+            medrecord (MedRecord): The MedRecord object containing the data.
+
+        Returns:
+            float: The calculated relative risk between the treatment and control
+                groups.
+
+        Raises:
+            AssertionError: If the preconditions are not met, indicating a potential
+                issue with group formation or subject count retrieval.
+        """
+        (
+            num_treat_true,
+            num_treat_false,
+            num_control_true,
+            num_control_false,
+        ) = self._compute_subject_counts(medrecord=medrecord)
+
+        return (num_treat_true / (num_treat_true + num_treat_false)) / (
+            num_control_true / (num_control_true + num_control_false)
```

Comment:
In some functions in this file there are assertions before possibly dividing by zero, in some cases there aren't. This should be consistent